src/run_cmd.js

```diff
@@ -0,0 +1,63 @@
+import path from 'path'
+import async from 'async'
+import child_process from 'child_process'
+import xtend from 'xtend'
+import {sh, shFlag} from './config'
+
+/**
+ * run a `package.json` script and the related pre- and postscript.
+ * @param {String} cwd - current working directory.
+ * @param  {Object} argv - command line arguments.
+ */
+export default function runCmd (cwd, argv) {
+  const scripts = argv._.slice(1)
+  const pkg = require(path.join(cwd, 'package.json'))
+
+  if (!scripts.length) {
+    const availableScripts = Object.keys(pkg.scripts || [])
+    console.log('Available scripts: ' + availableScripts.join(', '))
+    return
+  }
+
+  const env = xtend(process.env, {
+    PATH: [
+      path.join(cwd, 'node_modules/.bin'), process.env.PATH
+    ].join(path.delimiter)
+  })
+
+  async.mapSeries(scripts, (scriptName, cb) => {
+    const scripts = [
+      `pre${scriptName}`,
+      scriptName,
+      `post${scriptName}`
+    ]
+
+    const scriptCmds = scripts.map((scriptName) => {
+      return pkg.scripts[scriptName]
+    }).filter((scriptCmd) => {
+      return scriptCmd
```

Comment:
This should throw instead, no? npm does if you try to run a script that doesn't exist.